src/core.rs

```diff
@@ -40,14 +40,36 @@ impl<Head> HeadOrEq<Head> {
 }
 
 #[derive(Debug, Clone)]
-pub(crate) struct SpecializedPrimitive {
-    pub(crate) primitive: PrimitiveWithId,
-    pub(crate) input: Vec<ArcSort>,
-    pub(crate) output: ArcSort,
+pub struct SpecializedPrimitive {
+    primitive: PrimitiveWithId,
+    input: Vec<ArcSort>,
+    output: ArcSort,
+}
+
+impl SpecializedPrimitive {
+    /// Get the name of this primitive
+    pub fn name(&self) -> &str {
+        self.primitive.0.name()
+    }
+
+    /// Get the output sort of this primitive
+    pub fn output(&self) -> &ArcSort {
+        &self.output
+    }
+
+    /// Get the input sorts of this primitive
+    pub fn input(&self) -> &[ArcSort] {
+        &self.input
+    }
+
+    /// Get the external function ID of this primitive
+    pub(crate) fn external_id(&self) -> ExternalFunctionId {
+        self.primitive.1
+    }
```

Comment:
Not that it really matters, but it seems a bit simpler to make these all pub and then make the external ID on the external function id `pub(crate)` to hide it?